base/compiler/methodtable.jl

```diff
@@ -92,12 +98,21 @@ upper bound of `sig`, or it is possible that among the upper bounds, there
 is no least element. In both cases `nothing` is returned.
 """
 function findsup(@nospecialize(sig::Type), table::InternalMethodTable)
+    return _findsup(sig, nothing, table.world)
+end
+
+function findsup(@nospecialize(sig::Type), table::OverlayMethodTable)
+    result = _findsup(sig, table.mt, table.world)
+    result === nothing || return result
+    return _findsup(sig, nothing, table.world) # fall back to the internal method table
+end
+
+function _findsup(@nospecialize(sig::Type), mt::Union{Nothing,Core.MethodTable}, world::UInt)
     min_valid = RefValue{UInt}(typemin(UInt))
     max_valid = RefValue{UInt}(typemax(UInt))
-    result = ccall(:jl_gf_invoke_lookup_worlds, Any, (Any, UInt, Ptr{Csize_t}, Ptr{Csize_t}),
-                   sig, table.world, min_valid, max_valid)::Union{MethodMatch, Nothing}
-    result === nothing && return nothing
-    (result.method, WorldRange(min_valid[], max_valid[]))
+    result = ccall(:jl_gf_invoke_lookup_worlds, Any, (Any, Any, UInt, Ptr{Csize_t}, Ptr{Csize_t}),
+                   sig, mt, world, min_valid, max_valid)::Union{MethodMatch, Nothing}
+    return result === nothing ? result : (result, WorldRange(min_valid[], max_valid[]))
```

Comment:
This is conservatively correct currently (since we return `Any`), but it would be more correct to return the tuple unconditionally, since the WorldRange applies to the lookup, regardless of the result.